hydromt_fiat/workflows/exposure_vector.py

```diff
@@ -267,24 +268,36 @@ def setup_roads(
             # add the function to segmentize the roads into certain segments
 
         # Add the Primary Object Type and damage function, which is currently not set up to be flexible
-        roads["Primary Object Type"] = "roads"
-        roads["Damage Function: Structure"] = "roads"
+        roads["Primary Object Type"] = "road"
+        roads["Damage Function: Structure"] = "road"
 
         self.logger.info(
-            "The damage function 'roads' is selected for all of the structure damage to the roads."
+            "The damage function 'road' is selected for all of the structure damage to the roads."
         )
         # Clip road to model boundaries
         roads = roads.clip(self.region)
 
+        # Convert OSM road from meters to feet (if model unit is feet)
+        road_length = get_road_lengths(roads)
+        if self.unit == Units.feet.value and str(source).upper() == "OSM":
+            road_length = road_length * Conversion.meters_to_feet.value
+        road_length = road_length.apply(lambda x: f"{x:.2f}")
+
+        # Add the max potential damage and the length of the segments to the roads
         if isinstance(road_damage, str):
-            # Add the max potential damage and the length of the segments to the roads
             road_damage = self.data_catalog.get_dataframe(road_damage)
-            roads[["Max Potential Damage: Structure", "Segment Length [m]"]] = (
+            roads[["Max Potential Damage: Structure", "Segment Length", "Extraction Method"]] = (
                 get_max_potential_damage_roads(roads, road_damage)
             )
+            roads["Extraction Method"] = "centroid"
         elif isinstance(road_damage, int):
-            roads["Segment Length [m]"] = get_road_lengths(roads)
+            roads["Segment Length"] = road_length
             roads["Max Potential Damage: Structure"] = road_damage
+            roads["Extraction Method"] = "centroid"
+        elif isinstance(road_damage, float):
+            roads["Segment Length"] = road_length
+            roads["Max Potential Damage: Structure"] = math.nan
+            roads["Extraction Method"] = "centroid"
```

Comment:
this part is not clear to me. For example why is this part `roads["Extraction Method"] = "centroid"` in the if statement since it is exactly the same every time?

Then why for road damage, there are option for it to be str, int or float? What does this mean? I think these functions need a bit of documentation as well.

Reply:
> this part is not clear to me. For example why is this part `roads["Extraction Method"] = "centroid"` in the if statement since it is exactly the same every time?
> 
> Then why for road damage, there are option for it to be str, int or float? What does this mean? I think these functions need a bit of documentation as well.

In this case I did not change the code that was already existing, it was written like this before by someone else (with the options for int, float, str). I guess it is to cover all possibilities of user input road data? I put merged the int, float in one elif statement to only distinguish between str and (int,float) and moved the extraction method out of the if-statements